stable/prometheus/templates/server-deployment.yaml

```diff
@@ -25,6 +25,8 @@ spec:
         component: "{{ .Values.server.name }}"
         release: {{ .Release.Name }}
     spec:
+      securityContext:
+        runAsUser: 0
```

Comment:
Why is this necessary?

Reply:
NM - Just read earlier comments

Reply:
I have the following issue on my test cluster

```
  Warning  FailedCreate  3m (x17 over 8m)  replicaset-controller  Error creating: pods is forbidden: pod.Spec.SecurityContext.RunAsUser is forbidden
```

Reply:
I would rather remove this. Here's a possible solution
```
      initContainers:
      - name: "init-chown-data"
        image: "busybox"
        command: ["chown", "-R", "65534:65534", "{{ .Values.server.persistentVolume.mountPath }}"]
        volumeMounts:
        - name: storage-volume
          mountPath: {{ .Values.server.persistentVolume.mountPath }}
          subPath: "{{ .Values.server.persistentVolume.subPath }}"
```

Reply:
Thanks @lachie83 , I like that a lot better. Didn't realize that runAsUser could be disabled with `SecurityContextDeny`.

I'll make this change later today.

Reply:
Thanks. Otherwise I think it looks good. Once we get this merged, let's get prometheus 2.1 out there.